diff --git a/cegui/src/FactoryModule.cpp b/cegui/src/FactoryModule.cpp
index 7302a05d9..c1684f20e 100644
--- a/cegui/src/FactoryModule.cpp
+++ b/cegui/src/FactoryModule.cpp
@@ -28,6 +28,17 @@
 #include "CEGUI/FactoryRegisterer.h"
 #include "CEGUI/Exceptions.h"
 #include <stdio.h>
+
+// Fix problem of getWindowFactoryModule() not defined
+// as per http://cegui.org/forum/viewtopic.php?t=6473
+extern "C"{
+CEGUI::FactoryModule& getWindowFactoryModule()
+{
+    static CEGUI::FactoryModule mod;
+    return mod;
+}
+}
+
 // Start of CEGUI namespace section
 namespace CEGUI
 {
